.github/workflows/wheels.yml

```diff
@@ -131,18 +156,21 @@ jobs:
 
       - uses: actions/upload-artifact@5d5d22a31266ced268874388b861e4b58bb5c2f3 # v4.3.1
         with:
-          name: ${{ matrix.python }}-${{ startsWith(matrix.buildplat[1], 'macosx') && 'macosx' || matrix.buildplat[1] }}
+          name: ${{ matrix.python }}-${{ matrix.buildplat[1] }}-${{ matrix.buildplat[2] }}
           path: ./wheelhouse/*.whl
 
-      - uses: conda-incubator/setup-miniconda@030178870c779d9e5e1b4e563269f3aa69b04081 # v3.0.3
+      - uses: mamba-org/setup-micromamba@v1
         with:
           # for installation of anaconda-client, required for upload to
           # anaconda.org
-          # default (and activated) environment name is test
           # Note that this step is *after* specific pythons have been used to
           # build and test the wheel
-          auto-update-conda: true
-          python-version: "3.10"
+          # for installation of anaconda-client, for upload to anaconda.org
```

Comment:
changes here are incorporated from @rgommers work. The conda-incubator doesn't seem to work on some images.